src/frontend/token.cpp

```diff
@@ -261,6 +261,31 @@ namespace graphit {
         return *it;
     }
 
+    bool TokenStream::contains(Token::Type tokenType) const {
+        auto it = tokens.cbegin();
+        while(it != tokens.cend()){
+            Token foundToken = *it;
+            if (foundToken.type == tokenType) return true;
+            it++;
+        }
+        return false;
+
+    }
+
+    int TokenStream::findFirstOccurence(Token::Type tokenType) const {
```

Comment:
Add a comment here of what this utility function does?

Reply:
yep